packages/framework/src/Model/Feed/FeedExport.php

```diff
@@ -0,0 +1,195 @@
+use Symfony\Component\Filesystem\Filesystem;
+
+class FeedExport
+{
+    const TEMPORARY_FILENAME_SUFFIX = '.tmp';
+    const BATCH_SIZE = 1000;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Feed\FeedInterface
+     */
+    protected $feed;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig
+     */
+    protected $domainConfig;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Feed\FeedRenderer
+     */
+    protected $feedRenderer;
+
+    /**
+     * @var \Symfony\Component\Filesystem\Filesystem
+     */
+    protected $filesystem;
+
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    protected $em;
+
+    /**
+     * @var string
+     */
+    protected $feedFilepath;
+
+    /**
+     * @var int|null
+     */
+    protected $lastSeekId;
+
+    /**
+     * @var resource|null
+     */
+    protected $openedFile;
+
+    /**
+     * @var bool
+     */
+    protected $finished = false;
+
+    public function __construct(
+        FeedInterface $feed,
+        DomainConfig $domainConfig,
+        FeedRenderer $feedRenderer,
+        Filesystem $filesystem,
+        EntityManagerInterface $em,
+        string $feedFilepath,
+        ?int $lastSeekId
+    ) {
+        $this->feed = $feed;
+        $this->domainConfig = $domainConfig;
+        $this->feedRenderer = $feedRenderer;
+        $this->filesystem = $filesystem;
+        $this->em = $em;
+        $this->feedFilepath = $feedFilepath;
+        $this->lastSeekId = $lastSeekId;
+    }
+
+    public function generateBatch(): void
+    {
+        if ($this->finished) {
+            return;
+        }
+
+        $itemsInBatch = $this->feed->getItems($this->domainConfig, $this->lastSeekId, self::BATCH_SIZE);
+
+        if ($this->lastSeekId === null) {
+            $this->writeBegin();
+        }
+
+        $countInBatch = 0;
+        foreach ($itemsInBatch as $item) {
+            $this->writeItem($item);
+            $this->lastSeekId = $item->getSeekId();
+            $countInBatch++;
+        }
+
+        if ($countInBatch < self::BATCH_SIZE) {
+            $this->writeEnd();
+            $this->finishFile();
+        } else {
+            $this->closeFile();
+        }
+
+        $this->em->clear();
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Feed\FeedInfoInterface
+     */
+    public function getFeedInfo(): FeedInfoInterface
+    {
+        return $this->feed->getInfo();
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig
+     */
+    public function getDomainConfig(): DomainConfig
+    {
+        return $this->domainConfig;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getLastSeekId(): ?int
+    {
+        return $this->lastSeekId;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isFinished(): bool
+    {
+        return $this->finished;
+    }
+
+    protected function writeBegin(): void
```

Comment:
It was already in the `FeedXmlWriter` class. Methods `writeHeader` and `writeFooter` might be better - what do you think?

It would require renaming the Twig block in the feed templates. Is the change needed?

Reply:
Leave it as it is now.